pkg/routeagent_driver/handlers/calico/ippool_handler.go

```diff
@@ -240,65 +253,69 @@ func (h *calicoIPPoolHandler) updateROKSCalicoCfg(ctx context.Context) error {
+			}
 
-		if existing.Annotations == nil {
-			existing.Annotations = make(map[string]string)
-		}
+			if len(ipPools) == 0 {
+				logger.Infof("IPPools is empty, so nothing changed")
+				return existing, nil
+			}
 
-		// mark 'submariner-managed' label for persistency
-		existing.Labels[submarinerManagedLabel] = "true"
-		existing.Annotations[submarinerPrevIPIPMode] = string(existing.Spec.IPIPMode)
-		existing.Spec.IPIPMode = calicoapi.IPIPModeAlways
+			if installation.Annotations == nil {
+				installation.Annotations = map[string]string{}
+			}
 
-		logger.Infof("Setting IPIPMode of %s IPPool to Always", DefaultV4IPPoolName)
+			installation.Annotations[submarinerPrevEncapsulation] = ipPools[0].Encapsulation.String()
+			installation.Annotations[submarinerManagedLabel] = "true"
 
-		return existing, nil
-	})
+			ipPools[0].Encapsulation = tigerav1.EncapsulationIPIP
 
-	return errors.Wrap(err, "error updating default ippool")
+			return resource.MustToUnstructured(installation), nil
+		})
 }
 
 func (h *calicoIPPoolHandler) restoreROKSCalicoCfg() error {
-	err := util.Update(context.TODO(), h.iPPoolResourceInterface(), &calicoapi.IPPool{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: DefaultV4IPPoolName,
-		},
-	}, func(existing *calicoapi.IPPool) (*calicoapi.IPPool, error) {
-		prevIPIPModeCfg := existing.Annotations[submarinerPrevIPIPMode]
-		if prevIPIPModeCfg == "" {
-			return existing, nil // no need to update
-		}
-
-		existing.Spec.IPIPMode = calicoapi.IPIPMode(prevIPIPModeCfg)
-		delete(existing.Labels, submarinerManagedLabel)
-		delete(existing.Annotations, submarinerPrevIPIPMode)
-
-		return existing, nil
-	})
-
-	return errors.Wrap(err, "error updating default ippool")
-}
-
-func (h *calicoIPPoolHandler) iPPoolResourceInterface() resource.Interface[*calicoapi.IPPool] {
-	return &resource.InterfaceFuncs[*calicoapi.IPPool]{
-		GetFunc:    h.client.ProjectcalicoV3().IPPools().Get,
-		CreateFunc: h.client.ProjectcalicoV3().IPPools().Create,
-		UpdateFunc: h.client.ProjectcalicoV3().IPPools().Update,
-		DeleteFunc: h.client.ProjectcalicoV3().IPPools().Delete,
-	}
+	return util.Update(context.TODO(), resource.ForDynamic(h.dynClient.Resource(InstallationsGVR)),
+		resource.MustToUnstructured(&tigerav1.Installation{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: DefaultInstallationName,
+			},
+		}), func(existing *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+			installation := &tigerav1.Installation{}
+			err := runtime.DefaultUnstructuredConverter.FromUnstructured(existing.Object, installation)
+			utilruntime.Must(err)
+
+			prevEncapsulation := installation.Annotations[submarinerPrevEncapsulation]
+			if prevEncapsulation == "" {
+				return existing, nil // no need to update
+			}
+
+			ipPools := installation.Spec.CalicoNetwork.IPPools
+			if len(ipPools) > 0 {
+				ipPools[0].Encapsulation = tigerav1.EncapsulationType(prevEncapsulation)
+			}
+			delete(installation.Labels, submarinerManagedLabel)
```

Comment:
Linting error: "_expressions should not be cuddled with blocks_"

```suggestion
			
			delete(installation.Labels, submarinerManagedLabel)
```